Tests/TestUtilities/JavaProperties.cs

```diff
@@ -0,0 +1,165 @@
+﻿/*
+ * SonarQube Scanner for MSBuild
+ * Copyright (C) 2015-2017 SonarSource SA and Microsoft Corporation
+ * mailto: contact AT sonarsource DOT com
+ *
+ * Licensed under the MIT License.
+ * See LICENSE file in the project root for full license information.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TestUtilities
+{
+    /// <summary>
+    /// Hold Java style properties as key-value pairs and allow them to be loaded from or
+    /// saved to a ".properties" file. The file is stored with character set ISO-8859-1 which extends US-ASCII
+    /// (the characters 0-127 are the same) and forms the first part of the Unicode character set.  Within the
+    /// application <see cref="string"/> are Unicode - but all values outside the basic US-ASCII set are escaped.
+    /// </summary>
+    /// Copied from https://github.com/Kajabity/Kajabity-Tools
+    public class JavaProperties : Hashtable
```

Comment:
We have generics in C# so we usually never go for inheriting a collection. If you really need to, then I suggest that you inherit from Dictionary<TKey, TValue>.